zrcs/zrc-2.md

```diff
@@ -0,0 +1,372 @@
+
+
+#### 9. IsOperatorFor
+
+```ocaml
+(* @dev: Returns true if an address is an operator of tokenHolder. All addresses are their own operator. *)
+(* @param operator:     Address of a potential operator.                                                 *)
+(* @param tokenHolder:  Address of a token holder.                                                       *)
+transition IsOperatorFor(operator: ByStr20, tokenHolder: ByStr20)
+```
+
+|        | Name          | Type      | Description                          |
+| ------ | ------------- | --------- | ------------------------------------ |
+| @param | `operator`    | `ByStr20` | Address of a potential operator.     |
+| @param | `tokenHolder` | `ByStr20` | Address of a token ownwer.           |
+
+|           | Name                       | Description                             | Event Parameters                                                                                                                                                                                                               |
+| --------- | -------------------------- | --------------------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ |
+| eventName | `IsOperatorForSuccess` | Listing operators is successful.     | `operator`: `ByStr20`, and `tokenHolder`: `ByStr20`. |
+| eventName | `Error`                    | Listing operators is not successful. | TBA.                                                                                                      |
+
+
+#### 10. DefaultOperators
+
+```ocaml
+(* @dev: Returns the list of default operators. These addresses are operators for all token holders. *)
+transition DefaultOperators()
+```
+
+|           | Name                       | Description                             | Event Parameters                                                                                                                                                                                                               |
+| --------- | -------------------------- | --------------------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ |
+| eventName | `DefaultOperatorsSuccess` | Listing default operators is successful.     | TBA. |
+| eventName | `Error`                    | Listing default operators is not successful. | TBA.                                                                                                      |
+
+
+#### 11. Transfer
```

Comment:
I'm wondering if this should be called `Send` and the `Send` above should be called `Transfer`.

Reply:
I don't disagree that it would make more sense, but then there's also the `TransferFrom` transition. 

The potentially confusing part of using the ERC-777 standard is that it introduce operators but also supports the ERC-20 nomenclature. I merely followed the naming of their methods. I think for the sake of staying aligned with the ERC-777 naming convention, at least for now, we should stick the existing naming?

I could create a divide in the ZRC, indicating which part is the operator flow vs the erc20 flow?